### DISCUSSION TITLE
[Feature Request] Add backdrop

### DISCUSSION THREAD

**Discussion by vuxuanhungg** [ibhagwan/fzf-lua#1344] [Ideas]
`Lazy` plugin manager creates a backdrop right behind its main floating window. This helps to draw our attention to the main window. It would be nice if `fzf-lua` could adopt this idea.

#### Before:
![before](https://github.com/user-attachments/assets/38cbaabb-8c74-429a-aa38-d41faa53b7f7)

#### After:
![after](https://github.com/user-attachments/assets/0b7224a1-ed9d-4e63-acc2-199308b82793)

As you can see, the backdrop brings depth to the main floating window, thus creates a better UI/UX.

For demo, I create this little utility (which I "steal" from `lazy.nvim`) and call it before open `FzfLua`:
```lua
local create_backdrop = function()
  local backdrop_buf = vim.api.nvim_create_buf(false, true)
  local backdrop_win = vim.api.nvim_open_win(backdrop_buf, false, {
    relative = "editor",
    width = vim.o.columns,
    height = vim.o.lines,
    row = 0,
    col = 0,
    style = "minimal",
    focusable = false,
    zindex = 1,
  })
  local set_backdrop_win_opt = function(k, v)
    vim.api.nvim_set_option_value(k, v, {
      scope = "local",
      win = backdrop_win,
    })
  end

  vim.api.nvim_set_hl(0, "LazyBackdrop", { bg = "#000000", default = true })
  set_backdrop_win_opt("winhighlight", "Normal:LazyBackdrop")
  set_backdrop_win_opt("winblend", 60)
  vim.bo[backdrop_buf].buftype = "nofile"
  vim.bo[backdrop_buf].filetype = "lazy_backdrop"
end
```

**Comment by ibhagwan** [comment]
Added in https://github.com/ibhagwan/fzf-lua/commit/aaa2b267625d28d01375a9a26e6f8fd4df410d19, see the commits notes to customize, enabled by default with 60% opacity (same as Lazy).

**Reply by enoryw** [reply]
Such a feature is not compatible when enabling transparency for Neovim's colorscheme, right? E.g. with transparency enabled in [tokyonight.nvim](https://github.com/folke/tokyonight.nvim), it has no effect instead of a darker transparent (more opaque but not fully opaque) backdrop.

**Reply by ibhagwan** [reply]
> Such a feature is not compatible when enabling transparency for Neovim's colorscheme, right? E.g. with transparency enabled in [tokyonight.nvim](https://github.com/folke/tokyonight.nvim), it has no effect instead of a darker transparent (more opaque but not fully opaque) backdrop.

Disable it then with `winopts.backdrop`.

**Comment by vuxuanhungg** [comment]
What more can I say? Thank you!

**Reply by ibhagwan** [reply]
> What more can I say? Thank you!

You’re welcome!
